File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    var word = "" ;
    var ans = "" ;
    for (let i = 0 ; i < s.length ; i++) {
        if (s[i] == " ") {
            ans += reverse(word) ; 
            word = "" ;
            ans += " " ;
        } else {
            word += s[i] ;
        }
    }
    if (word != "") {
        ans += reverse(word) ;
    }
    return ans ;
};
var reverse = function(s) {
    if (s === "") {
        return "" ;
    } else {
        return reverse(s.substr(1)) + s.charAt(0) ;
    }
}

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        string word = "" , ans = "" ; 
        for (int i = 0 ; i < s.length() ; i++) {
            if (s[i] == ' ') {
                reverse(word.begin() , word.end()) ;
                ans += word ; 
                ans += " " ;
                word = "" ;
            } else {
                word += s[i] ;
            }
        }
        if (!word.empty()) {
            reverse(word.begin() , word.end()) ;
            ans += word ; 
        }
        return ans ;
    }
};